js/todolistview.js

```diff
@@ -0,0 +1,394 @@
+    // notify user that operation is failed.
+    alert("할 일을 제거하는 데 실패했습니다!");
+  }
+}
+
+// 할 일을 클릭했을때 이벤트 위임.
+function onTodoElementClicked(event) {
+  if (!event.target) {
+    return;
+  }
+
+  if (
+    event.target.nodeName === "INPUT" &&
+    event.target.classList.contains("toggle")
+  ) {
+    toggleTodoElementStatus(event);
+  } else if (event.target.nodeName === "BUTTON") {
+    removeCurrentTodoElement(event);
+  }
+}
+
+// 할 일을 더블클릭했을때 이벤트 위임.
+function onTodoElementDblclicked(event) {
+  if (event.target && event.target.nodeName === "LABEL") {
+    toggleTodoElementMode(event);
+  }
+}
+
+// 할 일을 조작 중 키를 입력했을때 이벤트 위임.
+function onTodoElementKeyupped(event) {
+  if (event.target && event.target.nodeName === "INPUT") {
+    updateTodoEdit(event);
+  }
+}
+
+// 할 일의 우선순위 변경.
+async function changeTodoElementPriority({ target }) {
+  const selectedPriority = parseInt(target.value); // 선택된 option 값은 select 요소의 value로 존재!
+  const userID = getActiveUserID();
+  const todoElement = target.closest("li");
+  const todoElementID = todoElement.id;
+
+  const animationToggler = getTodoElementLoadingAnimationToggler(todoElement);
+  animationToggler();
+
+  const updatedTodoElement = await updateTodoElementPriority(
+    userID,
+    todoElementID,
+    selectedPriority == 0 ? "NONE" : selectedPriority == 1 ? "FIRST" : "SECOND"
+  );
+  // 서버측 결과에 따라 스타일, 선택 항목 적용.
+  switch (updatedTodoElement.priority) {
+    case "FIRST":
+      target.value = 1;
+      target.classList.remove("secondary");
+      target.classList.add("primary");
+      break;
+    case "SECOND":
+      target.value = 2;
+      target.classList.remove("primary");
+      target.classList.add("secondary");
+      break;
+    case "NONE":
+      target.value = 0;
+      target.classList.remove("primary");
+      target.classList.remove("secondary");
+  }
+
+  animationToggler();
+}
+
+// 할 일 완료 여부 설정/해제.
+async function toggleTodoElementStatus({ target }) {
+  const todoElement = target.closest("li");
+
+  const animationToggler = getTodoElementLoadingAnimationToggler(todoElement);
+  animationToggler();
+
+  const updatedTodoElement = await updateTodoElementStatus(
+    getActiveUserID(),
+    todoElement.id
+  );
+  // checked 속성을 toggle input에 넣어줄 필요가 없는듯?
+  if (updatedTodoElement.isCompleted) {
+    // target.setAttribute('checked', '')
+    todoElement.classList.add("completed");
+  } else {
+    // target.removeAttribute('checked')
+    todoElement.classList.remove("completed");
+  }
+  applySelectedFilter();
+  updateCountText();
+
+  animationToggler();
+}
+
+// 할 일을 더블클릭 했을 때 편집 모드 토글.
+function toggleTodoElementMode({ target }) {
+  target.closest("li").classList.toggle("editing");
+}
+
+// 할 일 변경 이벤트 처리기.
+async function updateTodoEdit({ target, key }) {
+  const todoElement = target.closest("li");
+  if (key === "Escape") {
+    todoElement.classList.toggle("editing");
+  } else if (key === "Enter") {
+    const newTodoText = target.value.trimStart().trimEnd();
```

Comment:
```suggestion
    const newTodoText = target.value.trim();
```
```trim()```메서드를 사용하게 되면 문자열의 앞, 뒤 공백을 모두 제거해주기 때문에 ```trimStart()```와 ```trimEnd()```를 모두 사용하는 것보다 ```trim()``` 단일 메서드를 사용하는게 더 좋을거 같습니다!

Reply:
훨씬 편리하네요. 감사합니다!

review#4 커밋에서 반영되었습니다.